internal/api/grpc_service.go

```diff
@@ -27,3 +28,121 @@ func (s *GrpcService) GetDeviceDefinitionByID(ctx context.Context, in *p_grpc.Ge
 
 	return result, nil
 }
+
+func (s *GrpcService) GetDeviceDefinitionByMMY(ctx context.Context, in *p_grpc.GetDeviceDefinitionByMMYRequest) (*p_grpc.GetDeviceDefinitionItemResponse, error) {
+
+	qryResult, _ := s.Mediator.Send(ctx, &queries.GetDeviceDefinitionByMakeModelYearQuery{
+		Make:  in.Make,
+		Model: in.Model,
+		Year:  int(in.Year),
+	})
+
+	dd := qryResult.(queries.GetDeviceDefinitionByMakeModelYearQueryResult)
+
+	result := &p_grpc.GetDeviceDefinitionItemResponse{
+		DeviceDefinitionId: dd.DeviceDefinitionID,
+		Name:               dd.Name,
+		ImageUrl:           dd.ImageURL,
+		Type: &p_grpc.GetDeviceDefinitionItemResponse_Type{
+			Type:  dd.Type.Type,
+			Make:  dd.Type.Make,
+			Model: dd.Type.Model,
+			Year:  uint32(dd.Type.Year),
+		},
+		VehicleData: &p_grpc.GetDeviceDefinitionItemResponse_VehicleInfo{
```

Comment:
I realized a number of these are strings that should likely be floats. Let's change it before we start using the contract and then changing them becomes very hard:
- MPGHighway, MPGCity, MPG, FuelTankCapacityGal => float
- NumberOfDoors => int